### ISSUE TITLE
Different training behavior using tfrecord produced by "build_image_data.py" and "download_and_convert_cifar10.py"

### ISSUE THREAD

**Issue by Sixigma** [tensorflow/models#3483]
### System information
- **What is the top-level directory of the model you are using**: slim and inception
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: source
- **TensorFlow version (use command below)**: 1.4.0
- **Bazel version (if compiling from source)**: 0.5.4
- **CUDA/cuDNN version**: 8.0.61
- **GPU model and memory**: NVIDIA Corporation Device 1b06
- **Exact command to reproduce**:

```
bazel build //inception:build_image_data
bazel-bin/inception/build_image_data \
--train_directory='.../cifar-10/train' \
--validation_directory='.../cifar-10/test' \
--labels_file='.../cifar-10/labels.txt' \
--train_shards=128 \
--validation_shards=8 \
--num_threads=8 \
--output_directory='.../cifar-10/tfrecord' 
```

### Describe the problem
Describe the problem clearly here. Be sure to convey here why it's a bug in TensorFlow or a feature request.

I tried different types of labels.txt when it comes to build_image_data.py : 
```
0 airplane
1 automobile
...
```

```
0 ( empty line )
1 airplane
2 automobile
...
```
and both with python script or bazel.

On the other hand , I also produced tfrecord by download_and_convert_cifar10.py .
Under the same hyper-parameters , I kickoff cifar10 with the later one , but tfrecord produced by build_image_data.py always diverge in the very beginning of training.

I check the cifar10 manually extracted , it's nothing weird .

Is there anything I missed ?
Is this right way to produce tfrecord with build_image_data.py ?
thank you.

**Comment by Sixigma** [comment]
I kickoff the training with hand made script modified from download_and_convert_flowers.py and here is the code :

```
if __name__ == '__main__':

      import dataset_utils

      tf.app.flags.DEFINE_string('train_dir', '', '')

      tf.app.flags.DEFINE_string('test_dir', '', '')

      tf.app.flags.DEFINE_string('output_dir', '', '')

      FLAGS = tf.app.flags.FLAGS

      train_filenames, train_class_names = _get_filenames_and_classes(FLAGS.train_dir)
      test_filenames, test_class_names = _get_filenames_and_classes(FLAGS.test_dir)
      train_class_names_to_ids = dict(zip(train_class_names, range(len(train_class_names))))
      test_class_names_to_ids = dict(zip(test_class_names, range(len(test_class_names))))

      _convert_dataset('train', train_filenames, train_class_names_to_ids,
                   FLAGS.output_dir)
      _convert_dataset('test', test_filenames, test_class_names_to_ids,
                   FLAGS.output_dir)

      labels_to_class_names = dict(zip(range(len(train_class_names)), train_class_names))
      dataset_utils.write_label_file(labels_to_class_names, FLAGS.output_dir)
```

I didn't go through build_image_data.py in detail , but that might something out of date.
